pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -504,3 +518,31 @@ func (ic *ifConfigurator) getInterceptedInterfaces(
 func (ic *ifConfigurator) getOVSInterfaceType() int {
 	return internalOVSInterfaceType
 }
+
+func (ic *ifConfigurator) addPostInterfaceCreateHook(containerID, endpointName string, containerAccess *containerAccessArbitrator, hook postInterfaceCreateHook) error {
+	if containerAccess == nil {
+		return fmt.Errorf("container lock cannot be null")
+	}
+	go func() {
+		err := wait.PollImmediate(time.Second, 60*time.Second, func() (done bool, err error) {
+			ifaceName := fmt.Sprintf("vEthernet (%s)", endpointName)
+			containerAccess.lockContainer(containerID)
+			defer containerAccess.unlockContainer(containerID)
+			if _, ok := ic.getEndpoint(endpointName); !ok {
+				klog.InfoS("HNSEndpoint doesn't exist in cache, exit goroutine", "HNSEndpoint Name", endpointName)
+				return true, nil
+			}
```

Comment:
Don't we need to check if endpoint ID equals like before?

Reply:
Greate catch. Originally I thought there should no cases that HNSEndpoint names are the same but IDs are different, so I remove that check. After a deep thinking, this might happen if kubelet recreate a Pod using the same name before the interface is configured. Then I will add the check back.